specification/appcomplianceautomation/resource-manager/Microsoft.AppComplianceAutomation/preview/2022-11-16-preview/appcomplianceautomation.json

```diff
@@ -0,0 +1,1374 @@
+          },
+          {
+            "$ref": "#/parameters/OfferGuidParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Success.",
+            "schema": {
+              "$ref": "#/definitions/SnapshotResourceList"
+            }
+          },
+          "default": {
+            "description": "Error response.",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v3/types.json#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        },
+        "x-ms-examples": {
+          "Snapshots_List": {
+            "$ref": "./examples/Snapshots_List.json"
+          }
+        }
+      }
+    },
+    "/providers/Microsoft.AppComplianceAutomation/reports/{reportName}/snapshots/{snapshotName}": {
+      "get": {
+        "tags": [
+          "AppComplianceAutomation"
+        ],
+        "description": "Get the AppComplianceAutomation snapshot and its properties.",
+        "operationId": "Snapshot_Get",
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/ReportNameParameter"
+          },
+          {
+            "$ref": "#/parameters/SnapshotNameParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Success.",
+            "schema": {
+              "$ref": "#/definitions/SnapshotResource"
+            }
+          },
+          "default": {
+            "description": "Error response.",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v3/types.json#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Snapshot_Get": {
+            "$ref": "./examples/Snapshot_Get.json"
+          }
+        }
+      }
+    },
+    "/providers/Microsoft.AppComplianceAutomation/reports/{reportName}/snapshots/{snapshotName}/download": {
+      "post": {
+        "tags": [
+          "AppComplianceAutomation"
+        ],
+        "description": "Download compliance needs from snapshot, like: Compliance Report, Resource List.",
+        "operationId": "Snapshot_Download",
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/ReportNameParameter"
+          },
+          {
+            "$ref": "#/parameters/SnapshotNameParameter"
+          },
+          {
+            "$ref": "#/parameters/ReportCreatorTenantIdParameter"
+          },
+          {
+            "$ref": "#/parameters/DownloadTypeParameter"
+          },
+          {
+            "$ref": "#/parameters/OfferGuidParameter"
+          }
```

Comment:
POST parameters should be in the body not in the query 

[https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/proxy-api-reference.md#resource-action-requests](https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/proxy-api-reference.md#resource-action-requests)

Reply:
@raosuhas

Thanks for your link.
But I have few questions.

1. Actually I saw there are a lot of RP, their API specs also using the query parameter in their POST action.
2. In this [Microsoft API guidance](https://microsoft.sharepoint.com/:p:/t/AzureAPIStewardshipBoard/EQZVvzdBQqlGoddNjiEc_jkB61r5G6HuDX03oum0254z6w?e=7l5uME). It also give a lot of sample of using query parameter for the POST actions.

Reply:
Hi @raosuhas ~
We moved some parameters to body now. only keep the in path parameters outside.
See [here](https://github.com/Azure/azure-rest-api-specs/pull/21152/files#diff-16cc0066db31d47ee8b4f0e9385ece034881677d74a14b4d48790444b24bda4dL467).